lib/Spot/Mapper.php

```diff
@@ -220,10 +220,20 @@ public function belongsTo(EntityInterface $entity, $foreignEntity, $localKey)
 
     /**
      * Prepare entity and load necessary objects on it
+     * @param EntityInterface $entity
+     * @return bool|null
      */
     public function prepareEntity(EntityInterface $entity)
     {
+        if (false === $this->eventEmitter()->emit('beforeLoad', [$entity, $this])) {
```

Comment:
I am not sure I am comfortable with this `beforeLoad` hook here, because technically the Entity is half loaded at this point - is has been hydrated with the property values from the database, but does not have any relations loaded. It seems like a `beforeLoad` hook would have to be fired before any values have loaded - e.g. essentially on a blank entity, though I'm not sure to what degree, because a completely blank entity would not likely be useful to have a hook for.

If you want to just remove this hook for now and leave the `afterLoad` hook in, we can give the `beforeLoad` hook some more consideration before releasing it as a feature -- unless you have any other good ideas on where to put it or what state the entity should be in when it fires.